src/Graphics/Glyphs/GlyphGeom.cc

```diff
@@ -1057,8 +1060,9 @@ void GlyphGeom::generateSuperEllipsoid(const Point& center, Tensor& t, double sc
 
   double cl = (eigvals[0] - eigvals[1]) / (eigvals[0] + eigvals[1] + eigvals[2]);
   double cp = 2.0 * (eigvals[1] - eigvals[2]) / (eigvals[0] + eigvals[1] + eigvals[2]);
-  double A = spow((1.0 - cl), emphasis);
-  double B = spow((1.0 - cp), emphasis);
+  bool linear = cl >= cp;
+  double A = linear ? spow((1.0 - cp), emphasis) : spow((1.0 - cl), emphasis);
```

Comment:
Some more specific critiques.

Don't do computation inside a ternary operator. They should be used sparingly, in simple expression. For this case, pull out the part that actually varies--the cp/cl variable--and put that in a ternary if you must, then do the computation once in code. Above all, try to ascribe to DRY--Don't Repeat Yourself.